checks/imap/imap_test.go

```diff
@@ -0,0 +1,47 @@
+package imap
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"github.com/emersion/go-imap/backend/memory"
+	"github.com/emersion/go-imap/client"
+	"github.com/emersion/go-imap/server"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStatusUp(t *testing.T) {
+
+	//Create new memory imap backend
+	go newImapServer()
+
+	// Arrange
+	c, err := client.Dial("localhost:1143")
+	require.NoError(t, err)
+
+	check := New(c)
+
+	// Act
+	err = check(context.Background())
+
+	// Assert
+	require.NoError(t, err)
+}
+
+func newImapServer() {
+	// Create a memory backend
+	be := memory.New()
```

Comment:
Nice that we can use an in-memory backend instead of a Docker image!

Reply:
👍